### DISCUSSION TITLE
Integrating Custom LLVM Pass with AFL++

### DISCUSSION THREAD

**Discussion by silvanjo** [AFLplusplus/AFLplusplus#1895] [Q&A]
My question is how to register a custom LLVM pass to AFL++.

I've developed an LLVM pass (using the legacy PassManager) and am looking to run it alongside the AFL++ passes. Despite registering it to the PassManager, it doesn't appear to be used by the AFL++ compiler wrapper.

**Comment by tokatoka** [comment]
because registering into PassManager doesn't mean it will automatically get run by the compiler wrapper.
you need to pass `-Xclang -load -Xclang <path to your llvm pass.so>`

**Comment by vanhauser-thc** [comment]
Sorry I thought I had answered this already but apparently not. afl++ uses the new pass manager if llvm supports it and you cannot mix old and new. You can use an old llvm version eg. 11 maybe 12, then what tokatoka wrote works
